hail/python/test/hailtop/aiotools/test_copy.py

```diff
@@ -416,3 +417,22 @@ async def test_copy_src_parts(copy_test_context):
 
     await expect_file(fs, f'{dest_base}file1', 'src/a/file1')
     await expect_file(fs, f'{dest_base}subdir/file2', 'src/a/subdir/file2')
+
+
+@pytest.mark.asyncio
+async def test_file_and_directory_error_with_slash(router_filesystem):
+    sema, fs, bases = router_filesystem
+
+    src_base = await fresh_dir(fs, bases, 'gs')
+    dest_base = await fresh_dir(fs, bases, 'gs')
+
+    await fs.create(f'{src_base}empty/')
+
+    async with await fs.create(f'{src_base}not-empty/') as f:
+        await f.write('foo'.encode('utf-8'))
+
+    with pytest.raises(FileNotFoundError):
+        await fs.copy(sema, Transfer(f'{src_base}empty/', dest_base.rstrip('/')))
```

Comment:
Might also be nice to check this works as expected when the empty file isn't at the toplevel.  Like, have src_base/a/{foo, empty/} and copy src_base/a and make sure foo copies but not empty/.